python/orca/src/bigdl/orca/learn/utils.py

```diff
@@ -208,11 +210,25 @@ def convert_elem(elem):
                 return elem.tolist()
 
     def combine(pair):
-        return Row(*([pair[0][col] for col in pair[0].__fields__] +
-                     [convert_elem(pair[1])]))
+        if not output_cols:
+            if isinstance(pair[1], (list, tuple)) and len(pair[1]) == 1:
```

Comment:
what does this indicate? A singleton list?

Reply:
Yes, if users do not specified a output_col name, all output in pair[1] should be stacked into one column.
Note that pair[1] is always a list, in which case:
1. if pair[1] is a singleton list, we should return [a]
2. if pair[1] is not a singleton list, we should stack them as [[a, b, c]]